packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,104 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type { AvailableUpdateInfo, UpdatesInfo, UseUpdatesCallbacksType } from './UseUpdates.types';
+import { currentlyRunning } from './UseUpdatesConstants';
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+// Manifest is of type "any" until we no longer support classic updates
+export const availableUpdateFromManifest = (manifest: any) => {
+  return manifest
+    ? {
+        updateId: manifest?.id ?? null,
+        createdAt: manifest?.createdAt
+          ? new Date(manifest?.createdAt)
+          : manifest?.publishedTime
+          ? new Date(manifest?.publishedTime)
+          : null,
+        manifest,
+      }
+    : undefined;
+};
+
+// Constructs the UpdatesInfo from an event
+export const updatesInfoFromEvent = (
+  updatesInfo: UpdatesInfo | undefined,
+  event: UpdateEvent
+): UpdatesInfo => {
+  const lastCheckForUpdateTime = new Date();
+  if (event.type === Updates.UpdateEventType.NO_UPDATE_AVAILABLE) {
+    return {
+      ...updatesInfo,
+      currentlyRunning,
```

Comment:
no need to set this since it should already be in `updatesInfo` and is immutable. ditto with the ones below

Reply:
If I don't set `currentlyRunning`, I see Typescript errors -- it in fact is null briefly when the state is first created, I think

Reply:
Fixed